internal/js_library/js_library.bzl

```diff
@@ -17,8 +17,40 @@
 DO NOT USE - this is not fully designed, and exists only to enable testing within this repo.
 """
 
+_AMD_NAMES_DOC = """Mapping from require module names to global variables.
+This allows devmode JS sources to load unnamed UMD bundles from third-party libraries."""
+
+AmdNames = provider(
+    doc = "provide access to the amd_names attribute of js_library",
+    fields = {"names": _AMD_NAMES_DOC},
+)
+
+def write_amd_names_shim(actions, amd_names_shim, targets):
+  """Shim AMD names for UMD bundles that were shipped anonymous.
+
+  These are collected from our bootstrap deps (the only place global scripts should appear)
+
+  Args:
+    actions: skylark rule execution context.actions
+    amd_names_shim: File where the shim is written
+    targets: dependencies to be scanned for AmdNames providers
+  """
+
+  amd_names_shim_content = """// GENERATED by ts_web_test.bzl
+// Shim these global symbols which were defined by a bootstrap script
+// so that they can be loaded with named require statements.
+"""
+  for t in targets:
+    if AmdNames in t:
+      for n in t[AmdNames].names.items():
+        amd_names_shim_content += "define(\"%s\", () => %s);\n" % n
```

Comment:
Make this `define(\"%s\", function() { return %s; });` so its es5?

Reply:
yeah good idea to make it as portable as possible